### PR TITLE
some possible simplifications

### PR DESCRIPTION
**Pull request by nicolasaunai**

### CODE REVIEW

File: src/diagnostic/detail/h5writer.h [hunk 1]
```diff
@@ -269,7 +269,7 @@ namespace
         if (!file.exist(group_name))
             file.createGroup(group_name);
     }
-}
+} // namespace
```
Comment (PhilipDeegan) [inline src/diagnostic/detail/h5writer.h:272]: TORM

File: src/diagnostic/diagnostic_manager.h [hunk 2]
```diff
@@ -67,88 +78,77 @@ class DiagnosticsManager : public IDiagnosticsManager
...
+
 
+    auto& diagnostics() const { return diagnostics_; }
+
+
+    Writer& writer() { return *writer_.get(); }
+
+
+    DiagnosticsManager(const DiagnosticsManager&)  = delete;
+    DiagnosticsManager(const DiagnosticsManager&&) = delete;
```
Comment (PhilipDeegan) [inline src/diagnostic/diagnostic_manager.h:95]: redundant const?

File: src/diagnostic/detail/h5writer.h (hunk 1 above)
Comment (nicolasaunai) [reply src/diagnostic/detail/h5writer.h:272]: it's a clang format auto thing... can't erase, comes back

File: src/diagnostic/diagnostic_manager.h (hunk 2 above)
Comment (nicolasaunai) [reply src/diagnostic/diagnostic_manager.h:95]: I just moved these lines, didn't add anything. you suggest removing const in the && version? (I'd agree)

Comment (PhilipDeegan) [reply src/diagnostic/diagnostic_manager.h:95]: :man_shrugging:

Comment (nicolasaunai) [reply src/diagnostic/diagnostic_manager.h:95]: https://en.cppreference.com/w/cpp/language/move_constructor

File: src/diagnostic/diagnostic_manager.h [hunk 3]
```diff
@@ -67,88 +78,77 @@ class DiagnosticsManager : public IDiagnosticsManager
...
 
 
-    void addDiagnostic(DiagnosticProperties& diagnostic)
+    auto& diagnostics() const { return diagnostics_; }
+
+
+    Writer& writer() { return *writer_.get(); }
+
+
+    DiagnosticsManager(const DiagnosticsManager&) = delete;
```
Comment (PhilipDeegan) [inline src/diagnostic/diagnostic_manager.h:94]: "DiagnosticsManager const&" :)

File: src/diagnostic/diagnostic_writer.h [hunk 4]
```diff
@@ -15,15 +15,6 @@ class TypeWriter
     virtual ~TypeWriter() {}
```
Comment (PhilipDeegan) [inline src/diagnostic/diagnostic_writer.h:15]: is this not redundant also?

Comment (nicolasaunai) [reply src/diagnostic/diagnostic_writer.h:15]: you mean the abstract class as a whole?
it's actually used...

**Review by PhilipDeegan — APPROVED** [review]

**Pull request merged**